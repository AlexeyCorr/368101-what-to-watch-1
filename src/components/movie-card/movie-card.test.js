import React from 'react';
import renderer from 'react-test-renderer';

import MovieCard from './movie-card.jsx';

const mock =  {
  film: {
    picture: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`,
    link: `movie-page.html`
  }
};

it(`renders correctly`, () => {
  const {film} = mock;
  const clickHandler = jest.fn();
  const tree = renderer
    .create(<MovieCard
      film={film}
      clickHandler={clickHandler}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
