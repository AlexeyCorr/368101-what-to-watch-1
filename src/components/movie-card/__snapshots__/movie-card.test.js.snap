// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[MockFunction]}
  onMouseLeave={[MockFunction]}
>
  <video
    className="player__video"
    muted={true}
    poster="https://es31-server.appspot.com/wtw/static/film/preview/snatch.jpg"
  >
    <source
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      type="video/mp4"
    />
  </video>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/film/1"
      onClick={[Function]}
    >
      Snatch
    </a>
  </h3>
</article>
`;
